src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java

```diff
@@ -467,118 +873,162 @@ private JButton getPrintButton() {
+                    jTableInventoryRow.getColumnModel().getColumn(i).setPreferredWidth(0);
+                }
+            }
+            DefaultTableCellRenderer centerRenderer = new DefaultTableCellRenderer();
+            centerRenderer.setHorizontalAlignment(JLabel.CENTER);
+            jTableInventoryRow.getColumnModel().getColumn(3).setCellRenderer(centerRenderer);
             jTableInventoryRow.getSelectionModel().addListSelectionListener(new ListSelectionListener() {
 
                 @Override
                 public void valueChanged(ListSelectionEvent e) {
-                    if (!e.getValueIsAdjusting()) {
-                        jTableInventoryRow.editCellAt(jTableInventoryRow.getSelectedRow(), 5);
-                        jTetFieldEditor.selectAll();
+                    if (e.getValueIsAdjusting()) {
+                        jTableInventoryRow.editCellAt(jTableInventoryRow.getSelectedRow(), jTableInventoryRow.getSelectedColumn());
                     }
-
                 }
             });
-            jTableInventoryRow.getModel().addTableModelListener(new TableModelListener() {
 
-                @Override
-                public void tableChanged(TableModelEvent e) {
-
-                    if (e.getType() == TableModelEvent.UPDATE) {
-                        int row = e.getFirstRow();
-                        int column = e.getColumn();
-                        TableModel model = (TableModel) e.getSource();
-                        Object data = model.getValueAt(row, column);
-
-                        if (column == 2) {
-                            Object data2 = model.getValueAt(row, 3);
-                            if (!data.toString().equals("") && data2.toString().equals("")) {
-                                jTableInventoryRow.setSurrendersFocusOnKeystroke(true);
-                                jTableInventoryRow.getEditorComponent().requestFocus();
-                                return;
-                            }
-                        }
-
-                        if (column == 3) {
-                            Object data2 = model.getValueAt(row, 2);
-                            if (!data.toString().equals("") && data2.toString().equals("")) {
-                                jTableInventoryRow.setSurrendersFocusOnKeystroke(true);
-                                jTableInventoryRow.getEditorComponent().requestFocus();
-                                return;
-                            }
-                        }
-                    }
-                }
-            });
-            DefaultCellEditor cellEditor = new DefaultCellEditor(jTetFieldEditor);
+            DefaultCellEditor cellEditor = new DefaultCellEditor(jTextFieldEditor);
             jTableInventoryRow.setDefaultEditor(Integer.class, cellEditor);
         }
         return jTableInventoryRow;
     }
 
+    static class EnabledTableCellRenderer extends DefaultTableCellRenderer {
```

Comment:
What is the purpose of this?

The only method just calls the `super` method with exactly the same parameters.

Reply:
It's used to set the cell renderer for each column of the table

Reply:
Why not use the base class instead of creating a new one where the only method calls the super method with exactly the same parameters.   This is not needed.